### PR TITLE
Add data prep and evaluation scripts for puzzles

### PR DESCRIPTION
**Pull request by aszkiel71**
### Summary
This PR introduces the necessary infrastructure for fine-tuning and evaluating the actual model on chess puzzles. While the base model integration is still pending, the data pipeline and evaluation metrics are now fully functional.

### Changes
- **Rust Tooling**: Added `src/bin/prepare_puzzles.rs`. It streams the Lichess puzzle database (`.csv.zst`), filters for high-rated puzzles (Rating >= 2000), and outputs a clean `data/puzzles_filtered.jsonl` file.
- **Evaluation Script**: Added `scripts/evaluate_puzzles.py`. It implements a Pass@1 metric loop, integrating with the project's custom `Tokenizer`.
- **Future-proofing**: Included a `TokenFormat` enum to support research on different move representations (UCI, Enriched UCI, SAN).

### How to run
1. **Prep data**: `cargo run --release --bin prepare_puzzles`
2. **Run eval**: `uv run scripts/evaluate_puzzles.py` (currently uses mock inference until model is plugged in).

### Next Steps
- Replace the `predict_next_move` stub with the actual PyTorch forward pass once the model checkpoint is available.

### CODE REVIEW

File: src/prepare_puzzles.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        let fen = match record.get(1) {
+            Some(v) if !v.is_empty() => v.to_string(),
+            _ => {
+                eprintln!("Skipping record #{total}: missing FEN");
+                continue;
+            }
+        };
+
+        let first_move = match record.get(2).and_then(|m| m.split_whitespace().nth(1)) {
```
[the remark this answers is not in the document]
Comment (aszkiel71) [reply src/prepare_puzzles.rs]: dobra niech mu bedzie, moze bedzie mniej zamieszania w przyszlosci

**Comment by igorjakus** [comment]
> no nie ma bijekcji, ale przez lichess API to jest doable

a to git

**Comment by aszkiel71** [comment]
ogarne to potem, bo to jest kwestia dodania URL do partii, gdzie mozna łatwo wyciagnac UCI i spruningować do pozycji puzzle. after all trzeba sprawdzic czy ten fine tunning ma w ogóle sens, bo te partie mogly byc i tak w korpusie. 
(*może warto wyciac puzzlowy set z tego treningowego(?). ale still to lowkey pachnie jak overkill*)
